sway-lib-std/src/tx.sw

```diff
@@ -326,12 +336,9 @@ pub fn tx_script_data_length() -> Option<u64> {
 /// }
 /// ```
 pub fn tx_witnesses_count() -> u64 {
-    match tx_type() {
-        Transaction::Script => __gtf::<u64>(0, GTF_SCRIPT_WITNESSES_COUNT),
-        Transaction::Create => __gtf::<u64>(0, GTF_CREATE_WITNESSES_COUNT),
-        Transaction::Upgrade => __gtf::<u64>(0, GTF_SCRIPT_WITNESSES_COUNT),
-        Transaction::Upload => __gtf::<u64>(0, GTF_SCRIPT_WITNESSES_COUNT),
-        Transaction::Blob => __gtf::<u64>(0, GTF_SCRIPT_WITNESSES_COUNT),
+    match __gtf::<u8>(0, GTF_TYPE) {
```

Comment:
Not sure is cheaper, but I think we can do this in branchless mode with something like: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=dc5a8d77558dc968d29484a70afe485c

```
// GTF_TYPE = 0x001
// GTF_SCRIPT_WITNESSES_COUNT = 0x007
// GTF_CREATE_WITNESSES_COUNT = 0x105
// Something like this: as_u64(!(__gtf::<u8>(0, GTF_TYPE) ^ 1 > 0)) * (0x105-0x007) + 0x007
asm(t) {
    gtf t 0 i1; 
    xor t t $one;
    gt t t $zero;
    not t t
    and t t $one
    muli t t 0xFE
    addi t t 0x7
    gtf t 0 t
    t: u64
};
```

Reply:
Now this is the real gas golfing :smile: Thanks for the hint I am curious exploring it.